pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,354 @@
+
+
+@simulator_tracking
+@single_tape_support
+class LightningTensor(Device):
+    """PennyLane Lightning Tensor device.
+
+    A device to perform tensor network operations on a quantum circuit.
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+            Defaults to ``None`` if not specified.
+        backend (str): Supported backend. Currently, only ``quimb`` is supported.
+        method (str): Supported method. Currently, only ``mps`` is supported.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Currently, it can only be ``None``, so that computation of
+            statistics like expectation values and variances is performed analytically.
+        c_dtype: Datatypes for the tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is ``None``.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is ``1e-16``.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just consider the options for MPS simulator
+    _device_options = (
+        "apply_reverse_lightcone",
+        "backend",
+        "c_dtype",
+        "cutoff",
+        "method",
+        "max_bond_dim",
+        "measure_algorithm",
+        "return_tn",
+        "rehearse",
+    )
+
+    _new_API = True
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        wires=None,
+        backend="quimb",
+        method="mps",
+        shots=None,
+        c_dtype=np.complex128,
+        **kwargs,
+    ):
+
+        if not accepted_backends(backend):
+            raise ValueError(f"Unsupported backend: {backend}")
+
+        if not accepted_methods(method):
+            raise ValueError(f"Unsupported method: {method}")
+
+        if shots is not None:
+            raise ValueError("LightningTensor does not support the `shots` parameter.")
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._num_wires = len(self.wires) if self.wires else 0
+        self._backend = backend
+        self._method = method
+        self._c_dtype = c_dtype
+
+        # options for MPS
+        self._max_bond_dim = kwargs.get("max_bond_dim", None)
+        self._cutoff = kwargs.get("cutoff", 1e-16)
+        self._measure_algorithm = kwargs.get("measure_algorithm", None)
+
+        # common options (MPS and TN)
+        self._return_tn = kwargs.get("return_tn", False)
+        self._rehearse = kwargs.get("rehearse", False)
+        self._apply_reverse_lightcone = kwargs.get("apply_reverse_lightcone", None)
```

Comment:
Are you sure these are required for an MPS simulator? For instance, `rehearse` is used in ETNs to find a good contraction path, but with MPS there are usually known best algorithms/paths such that rehearsal is unnecessary. `apply_reverse_lightcone` is not applicable to MPS according to the Quimb [doc](https://github.com/jcmgray/quimb/blob/main/quimb/tensor/circuit.py#L3637-L3641), although it is overridden so it doesn't hurt.

Reply:
I created a story for discussing the kwargs that the used can pass to `LightningTensor` (we should potentially revisit the Notion page). In the meantime, I removed all of the above